internal/storage/bucket_handle_test.go

```diff
@@ -467,3 +469,67 @@ func (t *BucketHandleTest) TestUpdateObjectMethodWithMissingObject() {
 
 	AssertTrue(errors.As(err, &notfound))
 }
+
+func (t *BucketHandleTest) TestComposeObjectMethodWithValidObject() {
+	obj, err := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: TestObjectName,
+		})
+
+	AssertEq(nil, err)
+
+	composedObj, err := t.bucketHandle.ComposeObjects(context.Background(),
+		&gcs.ComposeObjectsRequest{
+			DstName:                       TestObjectName,
+			DstGenerationPrecondition:     nil,
+			DstMetaGenerationPrecondition: nil,
+			Sources: []gcs.ComposeSource{
+				{
+					Name: TestSubObjectName,
+				},
+			},
+			ContentType:     ContentType,
+			ContentEncoding: ContentEncoding,
+			ContentLanguage: ContentLanguage,
+			CacheControl:    CacheControl,
+			CustomTime:      CustomTime,
+			EventBasedHold:  true,
+			StorageClass:    StorageClass,
+			Metadata: map[string]string{
+				MetaDataKey: MetaDataValue,
+			},
+			Acl: nil,
+		})
+
+	AssertEq(nil, err)
+	AssertNe(nil, obj)
+	AssertGe(composedObj.Size, obj.Size)
```

Comment:
why are we not checking object properties are updated correctly or not

Reply:
fakeObject is not receiving any properties. so it is not updating any other properties.
https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/vendor/github.com/fsouza/fake-gcs-server/fakestorage/object.go#L833

Reply:
please compare the contents

Reply:
contents are still not compared?

Reply:
sorry didn't get you before. fixed